src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -222,6 +222,21 @@
   [max-frame-size url]
   (format "%s&wireFormat.maxFrameSize=%s&marshal=true" url max-frame-size))
 
+(defn- transfer-old-messages! [connection]
+  (let [[pending exists?]
```

Comment:
Looking at this again, I don't know that I'd change it to the existence call. I can see wanting to log the messages there before the messages have been moved.